sys/freebsd/evdev.txt

```diff
@@ -0,0 +1,178 @@
+define EVIOCSABS_MIN	EVIOCGABS(0xc0)
+define EVIOCSABS_MAX	EVIOCGABS(ABS_MAX+0xc0)
+define EVIOCGLED_MIN	EVIOCGLED(0)
+define EVIOCGLED_MAX	EVIOCGLED(LED_MAX)
+define EVIOCGKEY_MIN	EVIOCGKEY(0)
+define EVIOCGKEY_MAX	EVIOCGKEY(KEY_MAX)
+
+ioctl$EVIOCGVERSION(fd fd_evdev, cmd const[EVIOCGVERSION], arg int32)
+ioctl$EVIOCGID(fd fd_evdev, cmd const[EVIOCGID], arg ptr[out, input_id])
+ioctl$EVIOCGREP(fd fd_evdev, cmd const[EVIOCGREP], arg int32)
+ioctl$EVIOCSREP(fd fd_evdev, cmd const[EVIOCSREP], arg int32)
+ioctl$EVIOCGKEYCODE(fd fd_evdev, cmd const[EVIOCGKEYCODE], arg int32)
+ioctl$EVIOCGKEYCODE_V2(fd fd_evdev, cmd const[EVIOCGKEYCODE_V2], arg ptr[inout, input_keymap_entry])
+ioctl$EVIOCSKEYCODE(fd fd_evdev, cmd const[EVIOCSKEYCODE], arg int32)
+ioctl$EVIOCSKEYCODE_V2(fd fd_evdev, cmd const[EVIOCSKEYCODE_V2], arg ptr[in, input_keymap_entry])
+ioctl$EVIOCGABS(fd fd_evdev, cmd intptr[EVIOCGABS_MIN:EVIOCGABS_MAX], arg ptr[out, input_absinfo])
+ioctl$EVIOCSABS(fd fd_evdev, cmd intptr[EVIOCSABS_MIN:EVIOCSABS_MAX], arg ptr[in, input_absinfo])
+ioctl$EVIOCSFF(fd fd_evdev, cmd const[EVIOCSFF], arg ptr[in, ff_effect])
+ioctl$EVIOCRMFF(fd fd_evdev, cmd const[EVIOCRMFF])
+ioctl$EVIOCGEFFECTS(fd fd_evdev, cmd const[EVIOCGEFFECTS], arg int32)
+ioctl$EVIOCGRAB(fd fd_evdev, cmd const[EVIOCGRAB])
+ioctl$EVIOCREVOKE(fd fd_evdev, cmd const[EVIOCREVOKE])
+ioctl$EVIOCSCLOCKID(fd fd_evdev, cmd const[EVIOCSCLOCKID], arg int32)
+ioctl$UI_DEV_CREATE(fd fd_evdev, cmd const[UI_DEV_CREATE])
+ioctl$UI_DEV_DESTROY(fd fd_evdev, cmd const[UI_DEV_DESTROY])
+ioctl$UI_DEV_SETUP(fd fd_evdev, cmd const[UI_DEV_SETUP], arg ptr[in, uinput_setup])
+ioctl$UI_ABS_SETUP(fd fd_evdev, cmd const[UI_ABS_SETUP], arg ptr[in, uinput_abs_setup])
+ioctl$UI_GET_VERSION(fd fd_evdev, cmd const[UI_GET_VERSION], arg int32)
+ioctl$UI_SET_EVBIT(fd fd_evdev, cmd const[UI_SET_EVBIT])
+ioctl$UI_SET_KEYBIT(fd fd_evdev, cmd const[UI_SET_KEYBIT])
+ioctl$UI_SET_RELBIT(fd fd_evdev, cmd const[UI_SET_RELBIT])
+ioctl$UI_SET_ABSBIT(fd fd_evdev, cmd const[UI_SET_ABSBIT])
+ioctl$UI_SET_MSCBIT(fd fd_evdev, cmd const[UI_SET_MSCBIT])
+ioctl$UI_SET_LEDBIT(fd fd_evdev, cmd const[UI_SET_LEDBIT])
+ioctl$UI_SET_SNDBIT(fd fd_evdev, cmd const[UI_SET_SNDBIT])
+ioctl$UI_SET_FFBIT(fd fd_evdev, cmd const[UI_SET_FFBIT])
+ioctl$UI_SET_PHYS(fd fd_evdev, cmd const[UI_SET_PHYS])
+ioctl$UI_SET_SWBIT(fd fd_evdev, cmd const[UI_SET_SWBIT])
+ioctl$UI_SET_PROPBIT(fd fd_evdev, cmd const[UI_SET_PROPBIT])
+ioctl$UI_BEGIN_FF_UPLOAD(fd fd_evdev, cmd const[UI_BEGIN_FF_UPLOAD], arg ptr[inout, uinput_ff_upload])
+ioctl$UI_END_FF_UPLOAD(fd fd_evdev, cmd const[UI_END_FF_UPLOAD], arg ptr[in, uinput_ff_upload])
+ioctl$UI_BEGIN_FF_ERASE(fd fd_evdev, cmd const[UI_BEGIN_FF_ERASE], arg ptr[inout, uinput_ff_erase])
+ioctl$UI_END_FF_ERASE(fd fd_evdev, cmd const[UI_END_FF_ERASE], arg ptr[in, uinput_ff_erase])
+ioctl$UI_SET_BSDUNIQ(fd fd_evdev, cmd const[UI_SET_BSDUNIQ])
+ioctl$EVIOCGLED(fd fd_evdev, cmd intptr[EVIOCGLED_MIN:EVIOCGLED_MAX])
```

Comment:
For the ioctl taking `len` as an argument, I was not sure which len it's talking about. In syzkaller it asks for a type whose len it can take. I took the max value(So for EVIOCGLED I passed LED_MAX: https://github.com/freebsd/freebsd-src/blob/main/sys/dev/evdev/input.h#L98). Does that work? 
I will add the rest of them once I get the len part right.